src/SelfServe/SqlX/SqlX.tsx

```diff
@@ -50,30 +70,102 @@ export default class SqlX extends SelfServeBaseClass {
+
+    if (response.status != null)
+    {
+      if (!enableDedicatedGateway)
+      {
+        try 
+        {
+          await deleteDedicatedGatewayResource();
+          return { message: "DedicatedGateway resource will be deleted.", type: SelfServeNotificationType.info };
+        }
+        catch(e)
+        {
+          console.log(e);
+          return { message: "Deleting Dedicated Gateway resource failed. DedicatedGateway will not be deleted.", type: SelfServeNotificationType.error };
+        }
+      }
+      else
+      {
+        // Check for scaling up/down/in/out 
+      }
+    }
+    else
+    {
+      if (enableDedicatedGateway) {
+        const sku = currentValues.get("sku")?.value as string;
+        const instances = currentValues.get("instances").value as number;
+        try 
+        {
+          await updateDedicatedGatewayResource(sku, instances);
+          return { message: "Dedicated Gateway resource will be provisioned.", type: SelfServeNotificationType.info };
+        }
+        catch(e)
+        {
+          console.log(e);
+          return { message: "Updating Dedicated Gateway resource failed. Dedicated Gateway will not be updated.", type: SelfServeNotificationType.error };
+        }
+
+      }
+    }
+    return { message: "No updates were applied at this time", type: SelfServeNotificationType.warning };
   };
 
   public initialize = async (): Promise<Map<string, SmartUiInput>> => {
-    // TODO: get initialization data from initializeDedicatedGatewayProvisioning() RP call.
-    throw new Error("onSave not implemented");
+    // TODO: RP call to check if dedicated gateway has already been provisioned.
+    // Based on the RP call enableDedicatedGateway will be true if it has not yet been enabled and false if it has.
+    const defaults = new Map<string, SmartUiInput>();
+    const enableDedicatedGateway = false;
+    defaults.set("enableDedicatedGateway", { value: enableDedicatedGateway, hidden: false, disabled: false});
+    defaults.set("sku", { value: SKU.CosmosD4s, hidden: !enableDedicatedGateway, disabled: false});
+    defaults.set("instances", { value: await getInstancesMin(), hidden: !enableDedicatedGateway, disabled: false});    
+    const response = await getCurrentProvisioningState()
+    if (response.status != null)
+    {
+      disableAttributesOnDedicatedGatewayChange = true;
+      defaults.set("enableDedicatedGateway", { value: true, hidden: false, disabled: false});
+      defaults.set("sku", { value: response.sku, hidden: false, disabled: true});
+      defaults.set("instances", { value: response.instances, hidden: false, disabled: true});
+    }
+
+    return defaults;
   };
 
   @Values({
     description: {
-      textTKey: "Provisioning dedicated gateways for SqlX accounts.",
+      textTKey: "DedicatedGatewayDescription",
       link: {
         href: "https://docs.microsoft.com/en-us/azure/cosmos-db/introduction",
```

Comment:
change the link

Reply:
Okay, this one we don't have a link yet. We will add it closer to the Public Preview. I will add it to the backlog. THanks.